src/open_inwoner/configurations/bootstrap/zgw.py

```diff
@@ -1,500 +1,98 @@
-                "user_representation": org_label,
-                "server_certificate": server_certificate,
-            },
-        )
-
-    def test_configuration(self):
-        """
-        actual testing is done in final step
-        """
-
-
-class ZGWAPIsConfigurationStep(BaseConfigurationStep):
+    class Meta:
+        django_model_refs = {
+            OpenZaakConfig: [
+                "zaak_max_confidentiality",
+                "document_max_confidentiality",
+                "max_upload_size",
+                "skip_notification_statustype_informeren",
+                "reformat_esuite_zaak_identificatie",
+                "fetch_eherkenning_zaken_with_rsin",
+                "use_zaak_omschrijving_as_title",
+                "order_statuses_by_date_set",
+                "title_text",
+                "enable_categories_filtering_with_zaken",
+                "action_required_deadline_days",
+                "zaken_filter_enabled",
+            ]
+        }
+
+
+class OpenZaakConfigurationStep(BaseConfigurationStep):
     """
     Configure the ZGW settings and set any feature flags or other options if specified
     """
 
-    verbose_name = "ZGW APIs configuration"
-    config_settings = ConfigSettings(
-        enable_setting="ZGW_CONFIG_ENABLE",
-        namespace="ZGW",
-        models=[OpenZaakConfig],
-        related_config_settings=[
-            ZakenAPIConfigurationStep.config_settings,
-            CatalogiAPIConfigurationStep.config_settings,
-            DocumentenAPIConfigurationStep.config_settings,
-            FormulierenAPIConfigurationStep.config_settings,
-        ],
-        required_settings=[],
-        optional_settings=[
-            "ZGW_ACTION_REQUIRED_DEADLINE_DAYS",
-            "ZGW_ALLOWED_FILE_EXTENSIONS",
-            "ZGW_DOCUMENT_MAX_CONFIDENTIALITY",
-            "ZGW_ENABLE_CATEGORIES_FILTERING_WITH_ZAKEN",
-            "ZGW_FETCH_EHERKENNING_ZAKEN_WITH_RSIN",
-            "ZGW_MAX_UPLOAD_SIZE",
-            "ZGW_REFORMAT_ESUITE_ZAAK_IDENTIFICATIE",
-            "ZGW_SKIP_NOTIFICATION_STATUSTYPE_INFORMEREN",
-            "ZGW_TITLE_TEXT",
-            "ZGW_ZAAK_MAX_CONFIDENTIALITY",
-        ],
-    )
-
-    def is_enabled(self):
-        return getattr(settings, self.config_settings.enable_setting, False)
-
-    def is_configured(self) -> bool:
-        """Verify that at least 1 ZGW API set is configured."""
-        zgw_config = OpenZaakConfig.get_solo()
-        return ZGWApiGroupConfig.objects.filter(open_zaak_config=zgw_config).exists()
+    verbose_name = "Openzaak configuration"
+    enable_setting = "openzaak_config_enable"
+    namespace = "openzaak_config"
+    config_model = OpenZaakConfigurationModel
 
-    def configure(self):
-        if not self.is_enabled():
-            return
+    def execute(self, model: OpenZaakConfigurationModel):
+        if len(model.api_groups) < 1:
+            raise ConfigurationRunFailed("Configure at least one `api_groups` item")
```

Comment:
You could mention the configuration step to make clearer which part of the configuration is missing the api groups (unless that's already clear from the exception itself)